databuilder/extractor/snowflake_table_last_updated_extractor.py

```diff
@@ -0,0 +1,105 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+from pyhocon import ConfigFactory, ConfigTree
+from typing import Iterator, Union
+
+from databuilder import Scoped
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.sql_alchemy_extractor import SQLAlchemyExtractor
+from databuilder.models.table_last_updated import TableLastUpdated
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SnowflakeTableLastUpdatedExtractor(Extractor):
+    """
+    Extracts Snowflake table last update time from INFORMATION_SCHEMA metadata tables using SQLAlchemyExtractor.
+    Requirements:
+        snowflake-connector-python
+        snowflake-sqlalchemy
+    """
+    # TODO: SELECT statement from snowflake information_schema to extract table last update time
+    SQL_STATEMENT = """
+        SELECT
+            lower({cluster_source}) AS cluster,
+            lower(t.table_schema) AS schema,
+            lower(t.table_name) AS table_name,
+            DATA_PART(EPOCH, t.last_altered) AS last_updated_time
+        FROM
+            {database}.INFORMATION_SCHEMA.TABLES t
+        {where_clause_suffix};
+        """
+
+    # CONFIG KEYS
+    WHERE_CLAUSE_SUFFIX_KEY = 'where_clause_suffix'
+    CLUSTER_KEY = 'cluster_key'
+    USE_CATALOG_AS_CLUSTER_NAME = 'use_catalog_as_cluster_name'
+    # Database Key, used to identify the database type in the UI.
+    DATABASE_KEY = 'database_key'
+    # Snowflake Database Key, used to determine which Snowflake database to connect to.
+    SNOWFLAKE_DATABASE_KEY = 'snowflake_database'
```

Comment:
I found these confusing too. I just followed whatever was used in `snowflake_metadata_extractor.py`. `SNOWFLAKE_DATABASE_KEY` is cluster. Should have been cluster_key. DATABASE_KEY is same as here :https://github.com/amundsen-io/amundsendatabuilder/blob/1baec33899df8394b42b6a667ff2a5d1224b8528/databuilder/models/table_metadata.py#L209 ( would be `snowflake` here)

Reply:
i would suggest to consolidate and follow the existing metadata convention. thought?

Reply:
I thought about it too as the config keys are confusing but I wasn't how would it affect the existing users who are using it in production? Is it okay to make a backwards incompatible change and let users find it out when they run into issues?

Reply:
I need to take a look at the code when have time, but in general I think it will be better to consolidate and be consistent with other extractors.

Reply:
we could ship as it is and do a follow up pr.

Reply:
Sounds good. I would like to consolidate too(as long as it is not disruptive to folks using it). Follow up PR sounds good as it would be better to track the change that way